### PR TITLE
otel-cli: init at 0.0.20

### LINKED ISSUE #143471: Package Request: otel-cli
**Project description**
OTel-CLI is a command-line application to enhance existing shell based applications with instrumentation and observability via OTLP and OpenTelemetry

**Metadata**

* homepage URL: https://github.com/equinix-labs/otel-cli
* source URL: https://github.com/equinix-labs/otel-cli
* license: ASL2.0
* platforms: linux

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

Fixes #143471
Closes #143475

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/otel-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  preCheck = ''
+    ln -s $GOPATH/bin/otel-cli .
+    substituteInPlace main_test.go --replace \
+      'const minimumPath = `/bin:/usr/bin`' \
+      'const minimumPath = `'"$PATH"'`'
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/equinix-labs/otel-cli";
+    description = "a cli tool for otel instrumenting command line work";
```
Comment (deejayem) [inline pkgs/tools/misc/otel-cli/default.nix:27]: https://nixos.org/manual/nixpkgs/stable/#var-meta-description says "Capitalise the first character."

I think a clearer description would be "A command-line tool for sending OpenTelemetry traces" (from the beginning of the project README).

File: maintainers/maintainer-list.nix [hunk 2]
```diff
@@ -4115,6 +4115,12 @@
     githubId = 22533224;
     name = "Emma Bastås";
   };
+  emattiza = {
```
Comment (deejayem) [inline maintainers/maintainer-list.nix:4118]: When I added myself to this file, I was asked to do it as a separate commit (with message `maintainers: add NAME`). To be in alphabetical order, the entry should go before `emmabastas`, instead of after.

File: pkgs/tools/misc/otel-cli/default.nix (hunk 1 above)
Comment (deejayem) [reply pkgs/tools/misc/otel-cli/default.nix:27]: This doesn't appear to have changed.

Comment (urandom2) [reply pkgs/tools/misc/otel-cli/default.nix:27]: sorry, I had a number of changes I was wrangling today, this one got skipped, I will make sure to do this before resolving it again.

**Comment by deejayem** [comment]
Result of `nixpkgs-review pr 203582` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>otel-cli</li>
  </ul>
</details>

**Review by deejayem — COMMENTED** [review]
Ofborg is reporting test failures on darwin.

File: pkgs/tools/misc/otel-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  vendorHash = "sha256-IJ2Gq5z1oNvcpWPh+BMs46VZMN1lHyE+M7kUinTSRr8=";
+
+  nativeBuildInputs = [ getent ];
+
+  preCheck = ''
+    ln -s $GOPATH/bin/otel-cli .
+    substituteInPlace main_test.go --replace \
+      'const minimumPath = `/bin:/usr/bin`' \
+      'const minimumPath = `'"$PATH"'`'
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/otel-cli/default.nix:22]:
```suggestion
    substituteInPlace main_test.go \
      --replace 'const minimumPath = `/bin:/usr/bin`' 'const minimumPath = `'"$PATH"'`'
```

File: pkgs/tools/misc/otel-cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/equinix-labs/otel-cli";
+    description = "A command-line tool for sending OpenTelemetry traces";
+    changelog = "https://github.com/equinix-labs/${pname}/releases/tag/v${version}";
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/otel-cli/default.nix:28]:
```suggestion
    changelog = "https://github.com/equinix-labs/otel-cli/releases/tag/v${version}";
```

**Comment by urandom2** [comment]
> Ofborg is reporting test failures on darwin.

I can reproduce these failures locally without nix on x86_64-darwin, and I am pretty sure it is just the integration tests being too tightly coupled. There are several users that report that `otel-cli` works for them on darwin, and I can confirm the same about basic functionality. Since I was unable to find the exact failure, I will create an upstream issue and tempted to set `doCheck = !stdenv.IsDarwin`; thoughts?

**Comment by deejayem** [comment]
Result of `nixpkgs-review pr 203582` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>otel-cli</li>
  </ul>
</details>

**Comment by deejayem** [comment]
Result of `nixpkgs-review pr 203582` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>otel-cli</li>
  </ul>
</details>

**Comment by urandom2** [comment]
> Result of `nixpkgs-review pr 203582` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

This consistently reproduces differently on my `x86_64-darwin` system:
- `nixpkgs-review pr 203582`: fails
- `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`: works
- `nix shell .#otel-cli`

I know that macos has a very leaky sandbox, since it is usually disabled, so I am happy to take the feedback that this is broken, but it really seems like a bug in the tooling, not the derivation?
